src/python/pants/backend/codegen/protobuf/scala/rules.py

```diff
@@ -340,12 +328,23 @@ class PrefixedJvmJdkField(JvmJdkField):
     alias = "jvm_jdk"
 
 
+class PrefixedJvmResolveField(JvmResolveField):
```

Comment:
Is this because a `protobuf_source` might need a resolve for multiple languages?

Regardless, we may want to make a habit of putting these `Prefixed` classes in `jvm/rules.py` instead of duplicating these, just so that we don't have a proliferation of identical subclasses like this.

Reply:
It is because we need to know what resolve to search for the ScalaPB runtime library.

Reply:
These should be refactored in another PR to be in a common place for all JVM codegen.

Reply:
@tdyas Right, I'm just wondering why a field called `resolve` isn't enough here. If it's called `jvm_resolve`, that's because there might also be a `resolve` for Python?

Reply:
Yes. See `python_resolve` at https://github.com/pantsbuild/pants/blob/5db4b72f2cda68289204f0b2ebbe4e681df99914/src/python/pants/backend/codegen/thrift/apache/python/additional_fields.py#L11-L12.